### PR TITLE
Add support for Ubuntu (Fixes #19)

### LINKED ISSUE #19: Still can't get zeek-kafka to work
# Summary of the issue
zeek-kafka doesn't work, either via zkg or manual install

## Expected behavior
...

## Steps to reproduce

1) zkg install:
 zkg install seiso/zeek-kafka --version main
error: invalid package "seiso/zeek-kafka": package name not found in sources and also not a usable git URL (invalid or inaccessible, use -vvv for details)

2) manual install:
./configure --with-librdkafka=$LIBRDKAFKA_ROOT
make
sudo make install
zeek -N Seiso::Kafka
Seiso::Kafka - Writes logs to Kafka (dynamic, version 0.3.0)
zeek -r file.pcap /usr/local/zeek/share/zeek/site/local.zeek
fatal error in /usr/local/zeek/share/zeek/site/kafka.zeek, line 2: can't find packages/zeek-kafka/Seiso/Kafka
looking in /usr/local/zeek/share/zeek/site/packages, there is nothing there

*Where applicable, consider providing a patch that uses the [end to end testing environment](https://github.com/seisollc/zeek-kafka/blob/main/docker/run_end_to_end.sh).*

## Logs, errors, etc.
...

## Your environment
* Version of Zeek
* Version or commit hash of the zeek-kafka package
* Operating System and version

zeek 4.0.0-rc2
version 0.3
Ubuntu 20.04

### PR DESCRIPTION
**Pull request by JonZeolla**
# Summary of the contribution
Adds support for ubuntu:20.04

## Testing
`make e2e`

## Checklist
- [X] Confirm that any associated issues are indicated in the pull request title
- [X] Rebase your PR against the latest commit within the target branch
- [X] Include steps to verify and test the intended change
- [X] Write or update unit tests and/or integration tests to verify your changes
- [X] Run shellcheck against any new or updated shell scripts, indicating the reasoning behind any disabled checks
- [X] Indicate whether or not this is a breaking change
- [X] Ensure that any new dependencies are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)

### CODE REVIEW

**Comment by ottobackwards** [comment]
```bash
Complete!
Error: No matching repo to modify: powertools.
ERROR: Service 'zeek' failed to build : The command '/bin/sh -c dnf install -y 'dnf-command(config-manager)'  && dnf config-manager --set-enabled powertools' returned a non-zero code: 1
make: *** [e2e] Error 1
```

File: docker/containers/zeek/Dockerfile.ubuntu:20.04 [hunk 1]
```diff
@@ -14,60 +14,76 @@
...
-RUN dnf install -y epel-release
-
 # copy in the .screenrc
 COPY .screenrc /root
 
-# install powertools for libpcap-devel
-RUN dnf install -y 'dnf-command(config-manager)' && \
-    dnf config-manager --set-enabled powertools
+RUN DEBIAN_FRONTEND=noninteractive apt-get update \
+ && DEBIAN_FRONTEND=noninteractive apt-get -y install \
```
Comment (ottobackwards) [inline docker/containers/zeek/Dockerfile.ubuntu:20.04:29]: There should be tools parity between this and centos.  please add in vim

Comment (ottobackwards) [reply docker/containers/zeek/Dockerfile.ubuntu:20.04:29]: and jq and tree

Comment (JonZeolla) [reply docker/containers/zeek/Dockerfile.ubuntu:20.04:29]: You got it

**Comment by JonZeolla** [comment]
centos succeeds locally but is failing in the pipeline due to a disk space issue.  ubuntu is still an issue

**Comment by ckreibich** [comment]
Hi folks, as mentioned on the Zeek Slack I gave this a go with Ubuntu 20.10, which has [packages](https://packages.ubuntu.com/source/groovy/librdkafka) for librdkafka 1.4.2.  Here's a Dockerfile to get going:

```
FROM ubuntu:20.10

ENV DEBIAN_FRONTEND noninteractive

RUN apt update

# Install common dependencies 
RUN apt install -y bison bzip2 cmake curl flex g++ gcc git gpg libmaxminddb-dev \
    libpcap-dev libssl-dev make nullmailer python3 python3-dev python3-smmap \
    python3-gitdb python3-git python3-semantic-version swig tree zlib1g-dev

# Set up repo to install Zeek from SUSE OBS, as per
# https://software.opensuse.org//download.html?project=security%3Azeek&package=zeek
RUN echo 'deb http://download.opensuse.org/repositories/security:/zeek/xUbuntu_20.10/ /' | tee /etc/apt/sources.list.d/security:zeek.list
RUN curl -fsSL https://download.opensuse.org/repositories/security:zeek/xUbuntu_20.10/Release.key | gpg --dearmor | tee /etc/apt/trusted.gpg.d/security_zeek.gpg > /dev/null

RUN apt update
RUN apt install -y zeek
RUN echo 'export PATH=$PATH:/opt/zeek/bin' >/etc/profile.d/zeek.sh

# Install rdkafka
RUN apt install -y librdkafka-dev

ENTRYPOINT bash -l
```

When you run this and then say `LIBRDKAFKA_ROOT=/usr zkg install seisollc/zeek-kafka --version main`, you'll see it bails in the plugin build, with these errors:

```
  error in /opt/zeek/share/zeek/base/init-bare.zeek, line 1: cannot load plugin library /opt/zeek/var/lib/zkg/testing/zeek-kafka/clones/zeek-kafka/build//lib/SEISO-KAFKA.linux-x86_64.so: /opt/zeek/var/lib/zkg/testing/zeek-kafka/clones/zeek-kafka/build//lib/SEISO-KAFKA.linux-x86_64.so: undefined symbol: _ZN7RdKafka5Topic12PARTITION_UAE
```

Looks like it dislikes the use of `RdKafka::Topic::PARTITION_UA`. Not sure what's going on here since this doesn't happen when installing librdkafka 1.4.2 from https://github.com/edenhill/librdkafka/archive/refs/tags/v1.4.2.tar.gz.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -25,6 +25,8 @@ jobs:
     strategy:
       matrix:
         platform: [Ubuntu-20.04]
+        zeek-kafka-os: [ubuntu]
```
Comment (JonZeolla) [inline .github/workflows/ci.yml:28]: This suggestion should be applied once we iron out any "no space left on device" and Ubuntu build & issues.
```suggestion
        zeek-kafka-os: [centos, ubuntu]
```

**Comment by JonZeolla** [comment]
Still hitting out of disk errors, need to adjust the Zeek install process to use the open suse package...

**Comment by ckreibich** [comment]
One more comment — is `LIBRDKAFKA_ROOT` supposed to be the root of the installation tree, or the library folder? The default says `/usr/local/lib`, but it looks like the cmake checks look for things like `$LIBRDKAFA_ROOT/include/librdkafka/rdkafkacpp.h` ... so `/usr/local` might be a better default.

**Comment by JonZeolla** [comment]
Nice catch @ckreibich

**Comment by JonZeolla** [comment]
Ubuntu is now working locally.

**Comment by JonZeolla** [comment]
I'm not sure what to do about the disk issues yet.  It doesn't seem like the opensuse approach is compatible with our preferences on version-pinning packages - I'm only seeing the latest releases only on there.

**Comment by JonZeolla** [comment]
@ottobackwards had to remove the docker build caching but now e2e tests fit in GitHub actions 🤷‍♂️

**Comment by ottobackwards** [comment]
Well....   I really don't see us building often enough to get heart burn about the build times short of them timing out.
So, No worries from me.

**Comment by ottobackwards** [comment]
@JonZeolla conflicts need straightening out

**Comment by JonZeolla** [comment]
@ottobackwards rebased on main, should be good to go now.

**Review by ottobackwards — APPROVED** [review]

**Pull request merged**